mautic/mautic
# Add new Preference Center option to segments

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/5658
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added new option to segments - Editable by contact. This option allow select segments which will show in contact's preference center. Based on Alan's comments here  https://github.com/mautic/mautic/issues/5658#issuecomment-375363408 
This PR added also migration and set that new option with same values as Public option.

![image](https://user-images.githubusercontent.com/462477/43529740-35e8da80-95ac-11e8-8eb1-5fc486d67f65.png)

#### Steps to test this PR:
1.  Apply PR and run `php app/console doctrine:migrations:migrate`
2.  Go to segments and see segments with new option Editable by contact. Set some segments as Editable by contact
3. Go to preferences center in Mautic for contact and add all segments to contact.
4. Send email with unsubscribe link.
5. Open contact's preference center and see just segments with marked as editable
6. Change it and go to contact's profile in Mautic and see If just selected segments were removed.
